runtimes/protocol/lsp.ts

```diff
@@ -40,6 +41,47 @@ export class AutoParameterStructuresProtocolRequestType<P, R, PR, E, RO>
     }
 }
 
+/**
+ * Extra Client information passed during initialization handshake from Client to Server.
+ */
+export interface AWSClientInitializationOptions {
+    /**
+     * Custom client branding information.
+     */
+    product?: {
```

Comment:
Is this information a duplicate of the `clientInfo` in the `initializeRequest` itself? Can they be different?

Reply:
We can't be sure that standard `clientInfo` values set by IDEs can be customised by Client extensions or if values set by default are enough.
That's why I introduced this `product` field, where Client extensions can set values that are meaningful for them.